File: app/src/main/java/com/simplexray/an/OnConfigActionListener.java
package com.simplexray.an;

import java.io.File;
import java.util.concurrent.ExecutorService;

public interface OnConfigActionListener {
    void onEditConfigClick(File file);

    void onDeleteConfigClick(File file);

    ExecutorService getExecutorService();

    void switchVpnService();

    void createNewConfigFileAndEdit();

    void importConfigFromClipboard();

    void performBackup();

    void performRestore();

    void triggerAssetExtraction();

    void reloadConfig();
}

File: app/src/main/java/com/simplexray/an/OnConfigActionListener.kt
package com.simplexray.an

import java.io.File

interface OnConfigActionListener {
    fun onEditConfigClick(file: File?)

    fun onDeleteConfigClick(file: File?)

    fun switchVpnService()

    fun createNewConfigFileAndEdit()

    fun importConfigFromClipboard()

    fun performBackup()

    fun performRestore()

    fun triggerAssetExtraction()

    fun reloadConfig()
}
